### PR TITLE
[BI-1032] - Add Trait Entity and Attribute to Term Create/Update Form

### PR DESCRIPTION
**Pull request by dmeidlin**
The form for editing ontology terms was updated to include the changes in the Ontology MVP:

- required fields of Name, Description, Entity, Attribute, Method Description, Method Class, Scale Class with possible additional scale units and details;
- restyling summary details to show new and existing input fields as well as trait names, method names, and synonyms;
- archive link was removed since an archive slider switch is already in the edit form;
- the style was further improved for the Ontology MVP form for creating traits (`BaseTraitForm.vue`) to align more closely with the original wireframe; and
- the auto-completing of trait name and method name were moved into the respective heading instead of being shown as separate fields.

Three identified areas for future improvements to the style of `BaseTraitForm.vue` not addressed in this PR are:
- better responsiveness for tablet size viewport;
- correcting misalignment of field labels after validation error warnings appear; and
- preventing label text overlap for some scale detail inputs.

### CODE REVIEW

File: src/components/trait/forms/BaseTraitForm.vue [hunk 1]
```diff
@@ -1,166 +1,242 @@
...
+                v-on:input="trait.method.formula = $event"
+            />
+          </div>
+    </template>
+
+    <!-- Scale options -->
+    <template
+        v-if="trait.scale && (Scale.dataTypeEquals(trait.scale.dataType, DataType.Ordinal) || Scale.dataTypeEquals(trait.scale.dataType, DataType.Nominal))">
+      <div class="column is-2">
+        <span class="is-pulled-right">Categories</span>
```
Comment (nickpalladino) [inline src/components/trait/forms/BaseTraitForm.vue:203]: It'd be nice if there was a bit of space between the Categories label and the form input boxes.

**Review by nickpalladino — COMMENTED** [review]
New edit form looks good, couple small things.

File: src/components/tables/SidePanelTable.vue [hunk 2]
```diff
@@ -75,7 +75,7 @@
               v-show="records.length > 0"
               class="side-panel-scroll"
               v-if="sidePanelState.panelOpen"
-              v-bind:background-color-class="sidePanelState.editActive ? 'has-background-primary-light' : 'has-background-info-light'"
+              v-bind:background-color-class="sidePanelState.editActive ? 'has-background-primary-light' : 'has-background-light'"
```
[the remark this answers is not in the document]
Comment (dmeidlin) [reply src/components/tables/SidePanelTable.vue]: I changed the background color back to info-light and made the tag color info. I think that works better now, too.

File: src/components/trait/forms/BaseTraitForm.vue (hunk 1 above)
Comment (dmeidlin) [reply src/components/trait/forms/BaseTraitForm.vue]: Good catch! I fixed that and changed "Unit of Time" to just "Unit" for numerical scale class.

**Review by nickpalladino — APPROVED** [review]

**Review by timparsons — APPROVED** [review]

**Pull request merged**